### ISSUE TITLE
Migration aborted! Reason: Migration '20161012151407_xyz_content_type.yml' can not be ended as it is not executing

### ISSUE THREAD

**Issue by phalasz** [kaliop-uk/ezmigrationbundle#78]
Not sure what is causing this, but sometimes a migration file fails with the error mentioned in the title.

I'm using v2.1.0 of the bundle.

**Comment by phalasz** [comment]
A bit more information. It looks like it is trying to end the migration for a second time. Since it has already ended and has the correct end status it will fail.

The `.yml` definition had two instructions in there. One to create a folder and one to add the new content type.

```
Processing 20161012151407_xyz_content_type.yml
string(69) "Starting migration 20161012151407_xyz_content_type.yml"
string(85) "Migration started and added to DB. 20161012151407_xyz_content_type.yml"
string(67) "Ending Migration 20161012151407_xyz_content_type.yml"
array(6) {
  ["migration"]=>
  string(50) "20161012151407_xyz_content_type.yml"
  ["path"]=>
  string(87) "/app/src/AppBundle/MigrationVersions/20161012151407_xyz_content_type.yml"
  ["md5"]=>
  string(32) "4d1030968df48927ca35a1998a4cdd2a"
  ["execution_date"]=>
  string(10) "1476439712"
  ["status"]=>
  string(1) "1"
  ["execution_error"]=>
  NULL
}
string(67) "Migration ended. 20161012151407_xyz_content_type.yml"
string(67) "Ending Migration 20161012151407_xyz_content_type.yml"
array(6) {
  ["migration"]=>
  string(50) "20161012151407_xyz_content_type.yml"
  ["path"]=>
  string(87) "/app/src/AppBundle/MigrationVersions/20161012151407_xyz_content_type.yml"
  ["md5"]=>
  string(32) "4d1030968df48927ca35a1998a4cdd2a"
  ["execution_date"]=>
  string(10) "1476439712"
  ["status"]=>
  string(1) "2"
  ["execution_error"]=>
  NULL
}

Migration aborted! Reason: Migration '20161012151407_xyz_content_type.yml' can not be ended as it is not executing
```

**Comment by phalasz** [comment]
On some further investigation in my case there was an exception thrown, that was caught by https://github.com/kaliop-uk/ezmigrationbundle/blob/stable-2.x/Core/MigrationService.php#L226 and the subsequent call to `endMigration` which should set the migration to `failed` status will fail as the migration is already set to `done`.

The problem here is that the user executing is `anonymous` instead of a user with enough privileges to eZ's modules/functions.

**Comment by gggeek** [comment]
Fixed in 2.4 & 3.0beta2

**Comment by gggeek** [comment]
Fixed in 2.4 & 3.0beta2
